amadeus/booking/_hotel_orders.py

```diff
@@ -0,0 +1,45 @@
+from amadeus.client.decorator import Decorator
+
+
+class HotelOrders(Decorator, object):
+    def post(self,
```

Comment:
Makes sense, what do you think of using something like Pydantic to create data models?

Reply:
How would the data models help in this case? What I was thinking is the following. To allow developers call the API with 2 different ways:

**Option 1**
 ```    
amadeus.booking.hotel_orders.post(
        guests=guests, 
        travel_agent=travel_agent,
        room_associations=room_associations,
        payment=payment) 
```

**Option 2**
 ``` 
body = "value of body built by dev, in case they want to add some optional fields for example"
amadeus.booking.hotel_orders.post(body) 
```

In Java this could easily be achieved with an override function. In Python might be a bit more complex.

Reply:
Oh I see what you mean now. Yes it would be complex in Python as it does not support method overloading.

Slightly messy possibility:
```python3
    def post(self,
             guests=[],
             travel_agent={},
             room_associations=[],
             payment={},
             arrival_information={},
             body={}):
        if body:
            return self.client.post('/v2/booking/hotel-orders', body)
        else:
            # current implementation
            ...
```

Option 1 would work with this.
Option 2 would have to be modified to:
```
amadeus.booking.hotel_orders.post(body=body)
```

Would have to document that if `body` is present, then all other args would be ignored.

Reply:
Thanks @siddydutta. Then maybe we can leave it as it is, to avoid "messy" solutions in the codebase. And any developer can call the API using the arbitrary methods anyway by doing
```amadeus.post('/v2/booking/hotel-orders', body)```